### PR TITLE
Fix #53

### LINKED ISSUE #53: options.initial returns undefined when defined as empty string
Probably you should instead check of the existence of that property because currently, it sounds like it checks for falsey value.

```js
const res = await prompt({
  name: 'foo',
  type: 'input',
  message: 'Abc?',
  initial: ''
})
```

### PR DESCRIPTION
**Pull request by g-plane**
Fixes #53

### CODE REVIEW

File: lib/types/string.js [hunk 1]
```diff
@@ -180,6 +180,13 @@ class StringPrompt extends Prompt {
     this.write([header, prompt, footer].filter(Boolean).join('\n'));
     this.restore();
   }
+
+  set value(value) {
+    this.state.value = value;
+  }
+  get value() {
+    return [this.state.value, this.input].find(this.isValue.bind(this)) || this.initial;
+  }
```
Comment (jonschlinkert) [inline lib/types/string.js]: Hmm, shouldn't this be fixed [here](https://github.com/enquirer/enquirer/blob/master/lib/prompt.js#L361) instead?

Comment (jonschlinkert) [reply lib/types/string.js]: > That will influence other prompts like boolean prompt. (Of course I have tested.)

Right, that's the reason it's in the base `Prompt` class. Ideally, the logic should be simple enough that it works for every prompt. Of course, we can override it as you did in your PR if necessary. But first I want to try something... IMHO the code in your PR makes more sense than what's currently defined in `Prompt`. Before we merge in your changes, would you mind trying your code again in the `Prompt` class, but with the following code in the boolean prompt (and remove the code you added to the string prompt)?

```js
  set value(value) {
    super.value = value;
  }
  get value() {
    return this.cast(super.value);
  }
```

This way we're still using the `value` logic in the `Prompt` class, and if that changes this should still work.

Comment (jonschlinkert) [reply lib/types/string.js]: thanks for doing this, btw!

Comment (g-plane) [reply lib/types/string.js]: Updated.

**Comment by tunnckoCore** [comment]
Greaty.

**Pull request merged**